### PR TITLE
 Add sonarcloud and jacoco

### PR DESCRIPTION
**Pull request by Lucia560**
I addeed SonarCloud and jacoco .I  also removed 2 dependencies , web socket and Jax rs that where already included in the jakarta enterprise dependency.

### CODE REVIEW

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (marcusju23) [inline .github/workflows/sonarcloud.yml:1]: Remove reduntant space.

File: .github/workflows/sonarcloud.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=Lucia560_Jakarta_labb
+
```
Comment (marcusju23) [inline .github/workflows/sonarcloud.yml:40]: Remove redundant space from here and down.

File: pom.xml [hunk 3]
```diff
@@ -37,44 +52,57 @@
       <scope>test</scope>
     </dependency>
     <!-- Jakarta API for WebSocket -->
-    <dependency>
+   <!-- <dependency>
       <groupId>jakarta.websocket</groupId>
       <artifactId>jakarta.websocket-api</artifactId>
       <version>2.2.0-M1</version>
       <scope>test</scope>
-    </dependency>
```
Comment (marcusju23) [inline pom.xml:45]: Remove commented out code.

**Review by marcusju23 — CHANGES_REQUESTED** [review]
Great work! Some minor fix-ups then its good for merge! 👍

**Review by marcusju23 — APPROVED** [review]

**Review by marcusju23 — APPROVED** [review]
Great work Lucia! The choice of the branch name was good as it contributes in quickly identifying the associated issue. As a minor enhancement, consider explicitly stating the issue(s) you are closing in your pull request(s). Keep it up! 👍

**Review by TrungMa1 — APPROVED** [review]
Nicely done Lucia! Overall, seems well-structured.

**Pull request merged**